File: .tmp_py4013/batch_12_dump/3329_count_substrings_with_k_frequency_characters_ii.js
// LeetCode 3329 - Count Substrings With K-Frequency Characters II
// https://leetcode.com/problems/count-substrings-with-k-frequency-characters-ii/

var numberOfSubstrings = function(s, k) {
    const n = s.length;
    let ans = 0;
    for (let i = 0; i < n; i++) {
        const freq = new Array(26).fill(0);
        for (let j = i; j < n; j++) {
            freq[s.charCodeAt(j) - 97]++;
            let ok = false;
            for (const f of freq) if (f >= k) { ok = true; break; }
            if (ok) { ans += n - j; break; }
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check16/3329_count_substrings_with_k_frequency_characters_ii.rs
struct Solution;
// LeetCode 3329 - Count Substrings With K-Frequency Characters II
// https://leetcode.com/problems/count-substrings-with-k-frequency-characters-ii/

impl Solution {
    pub fn number_of_substrings(s: String, k: i32) -> i64 {
        let n = s.len();
        let sb = s.as_bytes();
        let mut ans = 0i64;
        for i in 0..n {
            let mut freq = [0i32; 26];
            for j in i..n {
                freq[(sb[j] - b'a') as usize] += 1;
                let ok = freq.iter().any(|&f| f >= k);
                if ok {
                    ans += (n - j) as i64;
                    break;
                }
            }
        }
        ans
    }
}

fn main() {}
